tests/bluetooth/bsim/mesh/src/mesh_test.c

```diff
@@ -567,6 +567,7 @@ bool bt_mesh_test_sync(uint32_t channel_id, uint16_t wait_sec)
 	uint32_t recv_msg;
 	int wait = wait_sec * MSEC_PER_SEC;
 
+	k_sleep(K_USEC(tm_get_phy_max_resync_offset()));
```

Comment:
If you still want this `k_sleep` to be here, then we could at least move `k_sleep` at line 529 to line 518 (before `bs_bc_is_msg_receive` call) and remove this `k_sleep`. This will basically do the same thing but more clean. Please also add a comment why sleep is called there.

But generally this will not solve the issue if we use the back channel directly in some other test. Probably we could use post init callback to sync devices at the test initialization, but not sure if this is correct.

Reply:
Back channel is not suitable functionality for interprocess synchronization at all. It has been sneaked somehow in mesh bsim tests for synchronization purposes and added undetermined behavior. As a right solution the synchronization via backchannel should be reverted and POSIX based approach should be implemented from the scratch.

Reply:
I'm not sure I got this as the back channel is POSIX based solution. We use the back channel not only for synchronization and this has already created a problem once. So the issue is not specific for synchronization.